src/futures_map.rs

```diff
@@ -127,15 +127,22 @@ where
     /// This iterator returns futures in an arbitrary order, which may change.
     ///
     /// If downcasting a future to `T` fails it will be skipped in the iterator.
-    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, &T)>
+    pub fn iter_of_type<T>(&self) -> impl Iterator<Item = (&ID, Pin<&T>)>
     where
         T: 'static,
     {
         self.inner.iter().filter_map(|a| {
             let pin = a.inner.inner.as_ref();
```

Comment:
Unsafe is not required for immutable references, because you can just use `get_ref` here, and return a `&T`:
https://doc.rust-lang.org/std/pin/struct.Pin.html#method.get_ref

Unfortunately unsafe is still required for mutable references, so you have to return a `Pin<&mut T>`.

Reply:
Are you saying we can just transmute the reference itself? Feel free to send a PR to this branch if you think it can be improved :)

Reply:
Not even a transmute:
```rust
            let pin = a.inner.inner.as_ref().as_ref();
```